### PR TITLE
Fix user profile info

### PR DESCRIPTION
**Pull request by joshuaobinna**

### CODE REVIEW

**Comment by Way2nnadi** [comment]
@joshuaobinna you need to run `npm run lint` before you push your code up for review. Not only that scan you PR and clean up your code

File: src/components/userProfile/components/UserAvatar.js [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
     STRINGS: { SQUARE },
     TEXTS: { BASIC_INFO_TEXT },
+    STRINGS: { FIRST_NAME, LAST_NAME, USER_NAME },
 } = USER_PAGE;
 
-export default class UserAvatar extends React.Component {
+class UserAvatar extends React.Component {
+
     render() {
+
```
Comment (Way2nnadi) [inline src/components/userProfile/components/UserAvatar.js:22]: nit deleted unused line

File: src/components/userProfile/components/UserAvatar.js [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+        const { firstName, lastName, username } = this.props.userData;
+
+        const BASIC_INFOMATION = [
+            { description: firstName, title: FIRST_NAME },
+            { description: lastName, title: LAST_NAME },
+            { description: username, title: USER_NAME },
+        ];
```
Comment (Way2nnadi) [inline src/components/userProfile/components/UserAvatar.js:29]: i do not agree with this pattern being here but its okay for now.

File: src/components/userProfile/components/UserAvatar.js [hunk 3]
```diff
@@ -24,10 +39,10 @@ export default class UserAvatar extends React.Component {
...
                         title={<h4 className={HEADER_TITLE}>{BASIC_INFO_TEXT}</h4>}
                         description={
-                            BASIC_INFOMATION.map(data => (
-                                <div key={data.title} className={BASIC_INFO}>
-                                    <b>{data.title}</b>
-                                    <div>{data.description}</div>
+                            BASIC_INFOMATION.map(({ description, title }) => (
+                                <div key={title} className={BASIC_INFO}>
+                                    {title}
+                                    <div>{description}</div>
```
Comment (Way2nnadi) [inline src/components/userProfile/components/UserAvatar.js:45]: remove `div`

File: src/components/userProfile/components/UserInfo.js [hunk 4]
```diff
@@ -19,27 +45,24 @@ export default class UserInfo extends React.Component {
...
-                        CONTACT.map(contact => (
-                            <div key={contact.title} className={DATA_TITLE}>
-                                <b>{contact.title}</b>
-                                <div>
-                                    {contact.description}
-                                </div>
+                        CONTACT.map(({ description, title }) => (
+                            <div key={title} className={DATA_TITLE}>
+                                {title}
+                                <div>{description}</div>
```
Comment (Way2nnadi) [inline src/components/userProfile/components/UserInfo.js:51]: remove unneeded `div`

File: src/components/userProfile/components/UserInfo.js [hunk 5]
```diff
@@ -19,27 +45,24 @@ export default class UserInfo extends React.Component {
...
-                        LOCATION.map(location => (
-                            <div key={location.title} className={DATA_TITLE}>
-                                <b>{location.title}</b>
-                                <div>
-                                    {location.description}
-                                </div>
+                        LOCATION.map(({ description, title }) => (
+                            <div key={title} className={DATA_TITLE}>
+                                {title}
+                                <div>{description}</div>
```
Comment (Way2nnadi) [inline src/components/userProfile/components/UserInfo.js:65]: remove unneeded `div`

File: src/components/userProfile/components/styles.css [hunk 6]
```diff
@@ -7,26 +7,26 @@
...
+    font-weight: 600;
 }
-.basic-info b {
-    font-size: 13.5px;
-}
+
 .basic-info div {
     float: right;
-    font-size: 13px;
+    font-size: 13.5px;
```
Comment (Way2nnadi) [inline src/components/userProfile/components/styles.css:27]: dont use decimal pixels. either leave it at `131 or move it up to another whole number

File: src/components/userProfile/sagas.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            yield effects.put(fetchUserData(user));
+        } else {
+            // eslint-disable-next-line no-console
+            console.log("User data not fetched");
+        }
+    } catch(error){
+        // eslint-disable-next-line no-console
+        console.log(error);
+    }
+}
```
Comment (Way2nnadi) [inline src/components/userProfile/sagas.js:24]: I am glad that you were able to figure this saga flow out, however, this is not needed given that the user information should already be accessible before you show the profile.

Comment (Way2nnadi) [reply src/components/userProfile/sagas.js:24]: in the app component there is a selector called `getUserData` use that to access user data not make another request to the server for data you already have

**Review by Way2nnadi — CHANGES_REQUESTED** [review]
Needs some medium level changes, but code looks clean overall

**Review by EngrEric — APPROVED** [review]
Looks good to me. Do the changes as requested by @enyeInc  and bump version

File: src/components/userProfile/components/UserAvatar.js (hunk 1 above)
Comment (joshuaobinna) [reply src/components/userProfile/components/UserAvatar.js:22]: Ok

File: src/components/userProfile/components/UserAvatar.js (hunk 3 above)
Comment (joshuaobinna) [reply src/components/userProfile/components/UserAvatar.js:45]: Div is used for styling

File: src/components/userProfile/components/UserInfo.js (hunk 4 above)
Comment (joshuaobinna) [reply src/components/userProfile/components/UserInfo.js:51]: Div is used for styling

File: src/components/userProfile/components/UserInfo.js (hunk 5 above)
Comment (joshuaobinna) [reply src/components/userProfile/components/UserInfo.js:65]: Div is used for styling

File: src/components/userProfile/components/styles.css (hunk 6 above)
Comment (joshuaobinna) [reply src/components/userProfile/components/styles.css:27]: Ok

File: src/components/userProfile/components/UserAvatar.js (hunk 2 above)
Comment (joshuaobinna) [reply src/components/userProfile/components/UserAvatar.js:29]: ok

**Pull request closed without merging**